instrumentation/active_model_serializers/lib/opentelemetry/instrumentation/active_model_serializers/instrumentation.rb

```diff
@@ -24,24 +32,39 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
           !defined?(::ActiveSupport::Notifications).nil? && gem_version >= MINIMUM_VERSION
         end
 
+        def subscribe
+          SUBSCRIPTIONS.each do |subscription_name|
+            OpenTelemetry.logger.debug("Subscribing to #{subscription_name} notifications with #{_tracer}")
+            OpenTelemetry::Instrumentation::ActiveSupport.subscribe(_tracer, subscription_name, default_attribute_transformer)
+          end
+        end
+
         private
 
+        def _tracer
+          self.class.instance.tracer
+        end
+
         def gem_version
           Gem::Version.new(::ActiveModel::Serializer::VERSION)
         end
 
-        def require_dependencies
-          require_relative 'event_handler'
+        def install_active_support_instrumenation
+          OpenTelemetry::Instrumentation::ActiveSupport::Instrumentation.instance.install({})
         end
 
-        def register_event_handler
-          ::ActiveSupport::Notifications.subscribe(event_name) do |_name, start, finish, _id, payload|
-            EventHandler.handle(start, finish, payload)
-          end
+        def require_dependencies
+          require_relative 'railtie'
```

Comment:
Should this _only_ be required if `Rails` is defined?

Reply:
As you [noticed](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1075#pullrequestreview-2188591330), the check for Rails being defined is within railtie.rb.